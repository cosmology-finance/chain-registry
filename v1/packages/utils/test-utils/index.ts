import { AssetList, Chain, IBCInfo } from '@chain-registry/types';

import _assets from '../../../../__fixtures__/v1/assets.json';
import _chains from '../../../../__fixtures__/v1/chains.json';
import _ibc from '../../../../__fixtures__/v1/ibc.json';

export const assets: AssetList[] = _assets as AssetList[];
// TODO update fixtures, they may be missing properties and/or old schema
export const chains: Chain[] = _chains as any;

export const ibc: IBCInfo[] = _ibc as IBCInfo[];
