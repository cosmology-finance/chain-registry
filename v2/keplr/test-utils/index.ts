import { AssetList, Chain, IBCData } from '@chain-registry/v2-types';
import { getFixturesRegistry } from '@chain-registry/workflows';

const fixturesReg = getFixturesRegistry();
// @ts-ignore
export const assets: AssetList[] = fixturesReg.assetLists
// TODO update fixtures, they may be missing properties and/or old schema
// @ts-ignore
export const chains: Chain[] = fixturesReg.chains;
// @ts-ignore
export const ibc: IBCData[] = fixturesReg.ibcData;
