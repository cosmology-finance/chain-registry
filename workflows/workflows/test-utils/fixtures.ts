import { join, resolve } from 'path';

import { Registry } from '../src/registry';

export type RegistryType = 'original' | 'minimal' | 'full'

export const getRegistry = (registryType: RegistryType = 'original') => {
  const registry = new Registry(
    join(__dirname, `/../../../repos/fixtures/registries/${registryType}`)
  );
  return registry;
};

export const fixtureOutputDir = resolve(__dirname + '/../../../__output__/');
