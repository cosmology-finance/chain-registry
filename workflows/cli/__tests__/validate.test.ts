import { CLI } from 'inquirerer';

import { commands } from '../src/commands/info';
import { setupTests, TestEnvironment } from '../test-utils';

const beforeEachSetup = setupTests();
const registryDir = '../../repos/fixtures/registries/original';

describe('Inquirerer', () => {
  let environment: TestEnvironment;

  beforeEach(() => {
    environment = beforeEachSetup();
  });

  it('prompts user and correctly processes delayed input', async () => {
    const { options  } = environment;

    const app = new CLI(commands, options, {
      _: ['validate'],
      registryDir
    });

    const result = await app.run();

    expect(result).toMatchSnapshot();
  });
});
