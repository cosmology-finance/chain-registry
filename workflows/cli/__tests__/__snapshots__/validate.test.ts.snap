// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inquirerer prompts user and correctly processes delayed input 1`] = `
{
  "_": [],
  "command": "validate",
  "registryDir": "../../repos/fixtures/registries/original",
  "tty": true,
}
`;
